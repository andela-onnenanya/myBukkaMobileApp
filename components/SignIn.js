import React,{Component} from 'react';
import { View,Text,TouchableOpacity,Animated,Image,Easing,ScrollView,TextInput } from 'react-native';
import styles,{colors} from '../styles/style'
import Inputs from './Inputs'
import propTypes from 'prop-types'
import Button from './Button'
import storage from '../data_Container/store'
import Img from './Images'
import lib from '../lib/lib'
import rsc,{mapStateToProps} from '../lib/resources'
import Swiper from 'react-native-swiper'
import {connect} from 'react-redux'

export default class SignIn extends Component{
    constructor(props){
        super(props)
        this.state={
            email:'',
            password:'',
            switchScreen:false
        }
        this.onLogin=this.onLogin.bind(this)
        this.onSend=this.onSend.bind(this)
    }
    onLogin(){
        if(this.state.email===''){
            alert('Email field cannot be empty')
        }else if(this.state.password===''){
            alert('Email field cannot be empty')
        }else{
            let email=this.state.email.toLowerCase(),
                password=this.state.password
            lib.signin(email,password)
        }
    }
    onSend(){
        if(this.state.email===''){
            alert('Email field cannot be empty')
        }else{
            let email=this.state.email
            lib.forgotPassword(email)
        }
    }
    render(){
        return(
        <View style={[styles.container,{backgroundColor:colors.a}]}>
            <View style={{flex:1}}></View>
            <View style={{flex:5,alignItems:'center',paddingLeft:10,paddingRight:10,justifyContent:'space-around'}}>
                <Img  
                    source={{uri:rsc.logo}} 
                    style={[{width:200,height:75,marginBottom:10}]}
                    onLoadStart={(e) => this.setState({loading: true})}
                    onLoad={(e) => this.setState({loading: false})}/>
                <ScrollView style={{width:'100%'}}>
                    {
                        (this.state.switchScreen)?
                        <View>
                            <Inputs     text="Email address"
                                        isPassword={false}
                                        onChangeText={(email)=>this.setState(()=>({email}))}
                                        value={this.state.email}
                                        />
                        </View>:
                        <View>
                            <Inputs     text="Email address"
                                        isPassword={false}
                                        onChangeText={(email)=>this.setState(()=>({email}))}
                                        value={this.state.email}
                                        />
                            <Inputs text="Password"
                                    isPassword={true}
                                    onChangeText={(password)=>this.setState(()=>({password}))}
                                    value={this.state.password}
                                    />
                        </View>
                        

                    }
                    
                    <TouchableOpacity
                        //style={{width:50,height:13}}
                        onPress={()=>this.setState((prevState)=>({switchScreen:!prevState.switchScreen}))}
                        >
                        <Text style={{paddingLeft:5,color:'rgba(255, 255, 255,1)',textAlign:'right',color:'#F69322'}}>
                            {(this.state.switchScreen)? 'Sign In?':'Forgot Password?'}
                        </Text>
                    </TouchableOpacity>
                    
                </ScrollView>
                {
                    (this.state.switchScreen)?
                    <Button text="Send"
                            textColor={[{color:colors.a}]}
                            event={this.onSend}
                            button={[{backgroundColor:'#fff',borderRadius:5, display:'flex',justifyContent:'center',alignItems:'center',padding:0,margin:0,shadowColor:'#000000',shadowRadius: 5,
                            shadowOpacity: .5,shadowOffset: {
                                width: 0,
                                height: 1
                            }},styles.button__Widec,styles.button__Long]}/>:
                    <Button text="Sign In"
                            textColor={[{color:colors.a}]}
                            event={this.onLogin}
                            button={[{backgroundColor:'#fff',borderRadius:5, display:'flex',justifyContent:'center',alignItems:'center',padding:0,margin:0,shadowColor:'#000000',shadowRadius: 5,
                            shadowOpacity: .5,shadowOffset: {
                                width: 0,
                                height: 1
                            }},styles.button__Widec,styles.button__Long]}/>
                }
            </View>
            <View style={{flex:1}}>
                <View style={{flex:1,justifyContent:'center',alignItems:'center'}}>
                    <Text style={{color:'rgba(255, 255, 255,.6)'}}>
                        Don't have an Account? 
                        <TouchableOpacity
                            style={{width:75,height:13}}
                            onPress={() => this.props.navigation.navigate('signup')}>
                            <Text style={{paddingLeft:5,color:'rgba(255, 255, 255,1)',fontWeight: 'bold'}}>
                                Register
                            </Text>
                        </TouchableOpacity>
                    </Text>
                </View>
            </View>
        </View>
        )
    }
}

