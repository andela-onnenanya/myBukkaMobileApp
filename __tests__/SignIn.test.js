import React from 'react'
import SignIn from '../components/SignIn'
import renderer from 'react-test-renderer'

test('sign in',()=>{
    const tree = renderer.create(<SignIn/>).toJSON()
    expect(tree).toMatchSnapshot();
})