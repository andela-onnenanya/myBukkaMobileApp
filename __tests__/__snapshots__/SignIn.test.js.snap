// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign in 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#9C1C26",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#9C1C26",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 5,
        "justifyContent": "space-around",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Image
      onLoad={[Function]}
      onLoadStart={[Function]}
      source={
        Object {
          "uri": "https://res.cloudinary.com/www-mybukka-com/image/upload/v1513294716/logo-light_xnxon0_rzghhp.png",
        }
      }
      style={
        Array [
          Object {
            "height": 75,
            "marginBottom": 10,
            "width": 200,
          },
        ]
      }
    />
    <RCTScrollView
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View>
        <View>
          <View
            style={
              Array [
                Object {
                  "width": "100%",
                },
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "height": 16,
                  },
                ]
              }
            />
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              onKeyPress={[Function]}
              placeholder="Email address"
              placeholderTextColor="#ffffff"
              style={
                Array [
                  Object {
                    "fontSize": 18,
                    "height": 50,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "borderBottomColor": "rgba(255,255,255,0.8)",
                    "borderBottomWidth": 1,
                    "width": "100%",
                  },
                  Object {
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "width": "100%",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "width": "100%",
                },
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "height": 16,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "borderBottomColor": "rgba(255,255,255,0.8)",
                    "borderBottomWidth": 1,
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                onKeyPress={[Function]}
                placeholder="Password"
                placeholderTextColor="#ffffff"
                secureTextEntry={true}
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                      "height": 50,
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                    Object {
                      "width": "90%",
                    },
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F69322",
                "paddingLeft": 5,
                "textAlign": "right",
              }
            }
          >
            Forgot Password?
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "margin": 0,
          "opacity": 1,
          "padding": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 5,
          "width": 158,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#9C1C26",
            },
          ]
        }
      >
        Sign In
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255,.6)",
          }
        }
      >
        Don't have an Account?
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 13,
              "opacity": 1,
              "width": 75,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255,1)",
                "fontWeight": "bold",
                "paddingLeft": 5,
              }
            }
          >
            Register
          </Text>
        </View>
      </Text>
    </View>
  </View>
</View>
`;
